graphql/src/Mu/GraphQL/Quasi.hs

```diff
@@ -0,0 +1,106 @@
+import           Language.GraphQL.Draft.Parser (parseSchemaDoc)
+import qualified Language.GraphQL.Draft.Syntax as GQL
+import           Language.Haskell.TH
+
+import           Mu.GraphQL.Annotations
+import           Mu.Rpc
+import           Mu.Schema.Definition
+
+-- | Imports an graphql definition written in-line as a 'Schema'.
+graphql :: String -> String -> T.Text -> Q [Dec]
+graphql scName svName schema =
+  case parseSchemaDoc schema of
+    Left e  -> fail ("could not parse graphql spec: " ++ show e)
+    Right p -> graphqlToDecls scName svName p
+
+data Result =
+    GQLSchema Type
+  | GQLService Type
+  deriving (Show, Eq)
+
+-- | Constructs the GraphQL tree splitting between Schemas and Services.
+graphqlToDecls :: String -> String -> GQL.SchemaDocument -> Q [Dec]
+graphqlToDecls schemaName serviceName (GQL.SchemaDocument types) = do
+  let schemaName'  = mkName schemaName
+      serviceName' = mkName serviceName
+  rs <- traverse (typeToDec schemaName') types
+  let schemaTypes  = [x | GQLSchema  x <- rs]
+      serviceTypes = [x | GQLService x <- rs]
+  schemaDec <- tySynD schemaName' [] (pure $ typesToList schemaTypes)
+  serviceDec <- tySynD serviceName' [] (pure $ typesToList serviceTypes)
+  pure [schemaDec, serviceDec]
+
+-- | Reads a GraphQL 'TypeDefinition' and returns a 'Result'.
+typeToDec :: Name -> GQL.TypeDefinition -> Q Result
+typeToDec _ (GQL.TypeDefinitionScalar (GQL.ScalarTypeDefinition _ s _)) = GQLSchema <$> scalarToType s
+  where
+    scalarToType :: GQL.Name -> Q Type
+    scalarToType (GQL.unName -> "Int")     = [t|'TPrimitive Int32|]
+    scalarToType (GQL.unName -> "Float")   = [t|'TPrimitive Double|]
+    scalarToType (GQL.unName -> "String")  = [t|'TPrimitive T.Text|]
+    scalarToType (GQL.unName -> "Boolean") = [t|'TPrimitive Bool|]
+    scalarToType (GQL.unName -> "ID")      = [t|'TPrimitive UUID|]
+    scalarToType _                         = fail "not well-known scalar types are not supported"
+typeToDec _ (GQL.TypeDefinitionObject objs) = objToDec objs
+  where
+    objToDec :: GQL.ObjectTypeDefinition -> Q Result
+    objToDec (GQL.ObjectTypeDefinition _ nm _ _ flds) =
+      GQLService <$> [t| 'Service $(textToStrLit $ GQL.unName nm) '[]
+          $(typesToList <$> traverse gqlFieldToType flds) |]
+    gqlFieldToType :: GQL.FieldDefinition -> Q Type
+    gqlFieldToType (GQL.FieldDefinition _ (GQL.unName -> fnm) args ftyp _) =
+      [t| 'Method $(textToStrLit fnm) '[]
+            $(typesToList <$> traverse argToType args)
+            'RetSingle $(gtypeToType ftyp) |]
+    argToType :: GQL.InputValueDefinition -> Q Type
+    argToType (GQL.InputValueDefinition _ (GQL.unName -> aname) atype Nothing) =
+      [t| 'ArgSingle ('Just $(textToStrLit aname)) '[] $(gtypeToType atype) |]
+    argToType (GQL.InputValueDefinition _ (GQL.unName -> aname) atype (Just defs)) =
+      [t| 'ArgSingle ('Just $(textToStrLit aname)) '[] {-$(fromGQLValueConst defs)-} $(gtypeToType atype) |]
```

Comment:
The default value should go as an annotation `DefaultValue`, in the list which is the second argument. [This line](https://github.com/higherkindness/mu-haskell/blob/master/graphql/exe/Main.hs#L59) has an example of how to use it.